sciencer/providers/provider.py

```diff
@@ -39,3 +39,14 @@ def get_papers_by_author(self, author_id: str = "") -> List[Paper]:
         Returns:
             List[Paper]: The list of papers associated with an author.
         """
+
+    @abstractmethod
+    def get_paper_by_terms(self, terms: List[Paper], max_papers: int) -> List[Paper]:
```

Comment:
Regarding the maximum number of papers, I was assuming exactly this: 
> Well but different providers might have lower thresholds...

I'll see if there is a cleaner way to fix this.

Reply:
Since users will mostly interact with Collectors rather than providers, I thinks its okay to keep it as is.

Maybe some code on the Provider to enforce some upper limit should be encouraged, even mandated by adding it to the interface itself.

That way you would have the Provider's limit (which is a implementation detail and the user shouldn't worry about it), and you'd have the Collector's limit which the user should use to limit the number of Papers to get as a response.

I'll resolve this since you've fixed the typing error, and the change I've envisioned would fall out of the scope of this pull request.